pandas/tseries/offsets.py

```diff
@@ -2145,8 +2166,41 @@ def onOffset(self, dt):
 
 def _tick_comp(op):
     def f(self, other):
-        return op(self.delta, other.delta)
+        if isinstance(other, Tick):
+            # Note we cannot just try/except other.delta because Tick.delta
+            # returns a Timedelta while Timedelta.delta returns an int
+            other_delta = other.delta
+        elif isinstance(other, (timedelta, np.timedelta64)):
+            other_delta = other
+        elif isinstance(other, Week) and other.weekday is None:
+            other_delta = timedelta(weeks=other.n)
+        elif isinstance(other, compat.string_types):
+            from pandas.tseries.frequencies import to_offset
+            other = to_offset(other)
+            if isinstance(other, DateOffset):
+                return f(self, other)
+            else:
+                if op == operator.eq:
+                    return False
+                elif op == operator.ne:
+                    return True
+                raise TypeError('Cannot compare type {self} and {other}'
+                                .format(self=self.__class__.__name__,
+                                        other=other.__class__.__name__))
+        elif op == operator.eq:
+            # TODO: Consider changing this older behavior for
+            # __eq__ and __ne__to match other comparisons
```

Comment:
what is this about?

Reply:
In the status quo `__eq__` returns `False` (and `__ne__` returns `True`) as the fallback when `other` cannot be converted to a `DateOffset`.  The other comparisons raise in this situation.  (I think Timedelta behaves the same way, so this might be OK).  This is just a note to double-check that we `__eq__` and `__ne__` should continue to be treated as special cases.